Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,573 @@
+
+    .. seealso::
+        [1] P. Klink, H. Abdulsamad, B. Belousov, C. D'Eramo, J. Peters, and J. Pajarinen,
+        "A Probabilistic Interpretation of Self-Paced Learning with Applications to Reinforcement Learning", arXiv, 2021
+    """
+
+    name: str = "sprl"
+
+    def __init__(
+        self,
+        env: DomainRandWrapper,
+        subroutine: Algorithm,
+        kl_constraints_ub,
+        max_iter: int,
+        performance_lower_bound: float,
+        std_lower_bound: float = 0.2,
+        kl_threshold: float = 0.1,
+        optimize_mean: bool = True,
+        optimize_cov: bool = True,
+        max_subrtn_retries: int = 1,
+    ):
+        """
+        Constructor
+
+        :param env: environment wrapped in a DomainRandWrapper
+        :param subroutine: algorithm which performs the policy/value-function optimization. Must also inherit from `ExposedSampler`
+        :param kl_constraints_ub: upper bound for the KL-divergence
+        :param max_iter: Maximal iterations for the SPRL algorithm (not for the subroutine)
+        :param performance_lower_bound: lower bound for the performance SPRL tries to stay above during distribution updates
+        :param std_lower_bound: clipping value for the standard deviation, necessary when using very small target variances
+        :param kl_threshold: threshold for the KL-divergence until which std_lower_bound is enforced
+        :param optimize_mean: whether the mean should be changed or considered fixed
+        :param optimize_cov: whether the (co-)variance should be changed or considered fixed
+        :param max_subrtn_retries: how often a failed (median performance < 30 % of performance_lower_bound) training attempt of
+                                   the subroutine should be reattempted
+        """
+        if not isinstance(subroutine, Algorithm):
+            raise pyrado.TypeErr(
+                given=subroutine,
+                expected_type=Algorithm,
+                msg="Subroutine must inherit from *Algorithm* and ExposedSampler!",
+            )
+        if not isinstance(subroutine, ExposedSampler):
+            raise pyrado.TypeErr(
+                given=subroutine,
+                expected_type=ExposedSampler,
+                msg="Subroutine must inherit from Algorithm and *ExposedSampler*!",
+            )
+        if not typed_env(env, DomainRandWrapper):
+            raise pyrado.TypeErr(given=env, expected_type=DomainRandWrapper)
+
+        # Call Algorithm's constructor with the subroutine's properties
+        super().__init__(subroutine.save_dir, max_iter, subroutine.policy, subroutine.logger)
+
+        # Using a Union here is not really correct, but it makes PyCharm's type hinting work suggest properties from both Algorithm and ExposedSampler.
+        self._subroutine: Union[Algorithm, ExposedSampler] = subroutine
+        self._subroutine.save_name = self._subroutine.name
+
+        self._env = env
```

Comment:
please group the attributes a bit nicer

Reply:
Done in f169acc2ee60dec843ef0cc1ca95c970fd518fdf